File: Tests/BLAS/MatrixMultDemo/blas-interfaces/blas_OpenBLAS_stubs.f90
! Stubs to use the OpenBLAS library
!

function blas_get_num_procs()

    implicit none

    integer :: blas_get_num_procs
    integer :: openblas_get_num_procs

    blas_get_num_procs = openblas_get_num_procs()

    return

end function blas_get_num_procs


subroutine blas_set_num_threads( num_procs )

    implicit none

    integer :: num_procs

    call openblas_set_num_threads( num_procs )

end subroutine blas_set_num_threads

File: Tests/BLAS/MatrixMultDemo/blas-interfaces/blas_OpenBLAS_stubs.c
extern  void openblas_set_num_threads(int num_threads) ;

#ifdef OLD_LIB

/* Old versions of OpenBLAS don't have the openblas_get_num_procs function. */

int blas_get_num_procs_(void) {
  return -1;
}

#else

extern int openblas_get_num_procs(void);

int blas_get_num_procs_(void) {
  return openblas_get_num_procs();
}

#endif

void blas_set_num_threads_(int* num_threads){
	openblas_set_num_threads(*num_threads);
}
